src/libcore/num/mod.rs

```diff
@@ -1416,18 +1416,14 @@ $EndFeature, "
 ```"),
             #[stable(feature = "no_panic_abs", since = "1.13.0")]
             #[rustc_const_stable(feature = "const_int_methods", since = "1.32.0")]
+            #[allow_internal_unstable(const_if_match)]
```

Comment:
I'm happy to accept the parts of the diff that does not use `#[allow_internal_unstable(...)]` but in my view, it is worse to use a stability hack than to use the existing hack as it makes us institutionally more relaxed about stability overall (and because there's a reason for why `const_if_match` is unstable, including that it has seen less vetting and testing).

Reply:
I see - that makes sense. I'll revert the changes to these bits that needed `allow_internal_unstable`, so the other changes can be merged in first.

I take it that we need `const_if_match` to stabilise first for these changes, then? I'll leave the original issue open and add a FIXME note, in that case.

Reply:
I think these changes are fine, as they don't add any new functionality except for reinstating our previous safety checks (assertions).

Reply:
Me and Oliver discussed this privately, we'll go with his idea for now.